src/servers/tracer.cc

```diff
@@ -181,9 +182,6 @@ TraceManager::TraceRelease(TRITONSERVER_InferenceTrace* trace, void* userp)
   if (parent_id == 0) {
     delete ts;
   }
-
-  LOG_TRITONSERVER_ERROR(
```

Comment:
Why are you removing this?

Reply:
TraceRelease is called in destructor of InferenceTrace by now. So it's no need to release it again. Otherwise, it will cause deadlock.

Reply:
I split original TraceRelease into two functions: TraceRelease and TraceStreamRelease.
TraceRelease should delete InferenceTrace object.
TraceStreamRelease should be called in InferenceTrace destructor.